concourse/pipelines/artifact-releaser-test.jsonnet

```
@@ -0,0 +1,423 @@
+// Imports.
+local common = import '../templates/common.libsonnet';
+
+// Get the repository given the build.
+local get_repo(build) = if std.startsWith(build, 'deb') then 'guest-arle-autopush-trusty'
+else if std.startsWith(build, 'el') then 'guest-arle-autopush-el7-x86_64'
+else 'guest-arle-autopush';
+
+// Get the universe given the build.
+local get_universe(build) = if std.startsWith(build, 'deb') then 'cloud-apt'
+else if std.startsWith(build, 'el') then 'cloud-yum'
+else 'cloud-yuck';
+
+local generatetimestamptask = {
+  task: 'generate-timestamp',
+  config: {
+    platform: 'linux',
+    image_resource: {
+      type: 'registry-image',
+      source: { repository: 'bash' },
+    },
+    outputs: [{ name: 'timestamp' }],
+    run: {
+      path: '/usr/local/bin/bash',
+      args: [
+        '-c',
+        'timestamp=$((${EPOCHREALTIME/./}/1000)); echo $(($timestamp/1000)) | tee timestamp/timestamp; echo $timestamp | tee timestamp/timestamp-ms',
+      ],
+    },
+  },
+};
+
+local upload_arle_autopush_task = {
+  local tl = self,
+
+  package_paths:: error 'must set package_paths in upload_arle_autopush_task',
+  topic:: 'projects/artifact-releaser-autopush/topics/gcs-guest-package-upload-autopush',
+  type:: error 'must set type in upload_arle_autopush_task',
+  universe:: error 'must set universe in upload_arle_autopush_task',
+  repo:: error 'must set repo in upload_arle_autopush_task',
+
+  task: 'upload_arle_autopush_task',
+  config: {
+    platform: 'linux',
+    image_resource: {
+      type: 'image-registry',
```

```suggestion
      type: 'registry-image',
```

I don't know if this would cause the validation to fail, but we should use the right type anyways.